### PR TITLE
completes api/topics, tested with error handling

### PR DESCRIPTION
**Pull request by andmikpar**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+beforeEach(() => seed(testData));
+
+describe('GET/api/topics', () => {
+  test('status 200, returns an array of object in correct format', () => {
+    return request(app)
+      .get('/api/topics')
+      .expect(200)
+      .then((result) => {
+        const topics = result.body;
+        expect(topics).toBeInstanceOf(Array);
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: is this line necessary if we're iterating through successfully in the next line?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        topics.forEach((topic) => {
+          expect(topic).toEqual(
+            expect.objectContaining({
+              slug: expect.any(String),
+              description: expect.any(String),
+            })
```
Comment (roisinoreilly) [inline __tests__/app.test.js:27]: if you're looping through an array in a test you need to make sure it's not empty or you'll get a false positive, could you add that to this test please

File: app/error-handlers.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const badPath = (request, response, next) => {
+  response.status(404).send({ msg: 'Not Found' });
+};
+
+const handle500errors = (err, request, response, next) => {
```
Comment (roisinoreilly) [inline app/error-handlers.js:5]: Nicely handled

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
Great job so far! Your code is neat and easy to read. Would you be able to add in a line to the 200 test to make sure it's not empty before you loop through it then send another quick PR after making the changes :)

File: app/controllers/app.controllers.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const { findTopics } = require('../models/app.models');
+
+const getTopics = (request, response, next) => {
+  findTopics()
+    .then((result) => {
+      response.status(200).send(result.rows);
+    })
+    .catch(next);
```
Comment (Pipe-Smoking-Rabbit) [inline app/controllers/app.controllers.js:8]: very clean!

File: app/controllers/app.controllers.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const { findTopics } = require('../models/app.models');
+
+const getTopics = (request, response, next) => {
+  findTopics()
+    .then((result) => {
+      response.status(200).send(result.rows);
```
Comment (Pipe-Smoking-Rabbit) [inline app/controllers/app.controllers.js]: slight issue here, the response array must be sent on a property key

i.e.  `.send({ topics: result.rows});` 

This is convention for sending responses so that the data is stored on semantic key names for the end users to access - when you change this make sure to also update the test to deconstruct out this object key

**Review by Pipe-Smoking-Rabbit — COMMENTED** [review]
Looks great, just one slight issue in the controller which needs changing - once that's been done go ahead and merge it!
